tests/Cluster.py

```diff
@@ -304,25 +349,67 @@ def launch(self, pnodes=1, unstartedNodes=0, totalNodes=1, prodCount=1, topo="me
                 cmdArr.append("--spcfc-inst-nodeos")
                 cmdArr.append(path)
 
-        # must be last cmdArr.append before subprocess.call, so that everything is on the command line
-        # before constructing the shape.json file for "bridge"
-        if topo=="bridge":
-            shapeFilePrefix="shape_bridge"
-            shapeFile=shapeFilePrefix+".json"
+        def createDefaultShapeFile(newFile, cmdArr):
             cmdArrForOutput=copy.deepcopy(cmdArr)
             cmdArrForOutput.append("--output")
-            cmdArrForOutput.append(shapeFile)
+            cmdArrForOutput.append(newFile)
             s=" ".join(cmdArrForOutput)
             if Utils.Debug: Utils.Print("cmd: %s" % (s))
             if 0 != subprocess.call(cmdArrForOutput):
-                Utils.Print("ERROR: Launcher failed to create shape file \"%s\"." % (shapeFile))
+                Utils.Print("ERROR: Launcher failed to create shape file \"%s\"." % (newFile))
                 return False
 
-            Utils.Print("opening %s shape file: %s, current dir: %s" % (topo, shapeFile, os.getcwd()))
-            f = open(shapeFile, "r")
-            shapeFileJsonStr = f.read()
+            Utils.Print("opening shape file: %s, current dir: %s" % (newFile, os.getcwd()))
+            f = open(newFile, "r")
```

Comment:
Since we're already making changes here, it would be nice to switch to the with/as idiom used elsewhere and dispense with the explicit close.

Reply:
ok